internal/stacks/stackruntime/internal/stackeval/component_instance.go

```diff
@@ -576,6 +576,11 @@ func (c *ComponentInstance) CheckModuleTreePlan(ctx context.Context) (*plans.Pla
 				}
 			}
 
+			// The instance is also upstream deferred if the for_each value for this instance is unknown.
+			if c.key == addrs.WildcardKey {
+				upstreamDeferred = true
```

Comment:
nit: Let's rename this, to just `deferred` or `deferPlan` something. It's not upstream deferred if the instances are unknown it's deferred because of its own config.